engine/common/requester/engine.go

```diff
@@ -234,19 +237,26 @@ func (e *Engine) addEntityRequest(entityID flow.Identifier, selector flow.Identi
 // Force will force the requester engine to dispatch all currently
 // valid batch requests.
 func (e *Engine) Force() {
-	count := uint(0)
-	for {
-		dispatched, err := e.dispatchRequest()
-		if err != nil {
-			e.log.Error().Err(err).Msg("could not dispatch requests")
-			return
-		}
-		if !dispatched {
-			e.log.Debug().Uint("requests", count).Msg("forced request dispatch")
-			return
+	// using Launch to ensure the caller won't be blocked
+	e.unit.Launch(func() {
+		// using atomic bool to ensure there is at most one caller would trigger dispatching requests
+		if e.isDispatching.CompareAndSwap(false, true) {
+			count := uint(0)
+			for {
+				dispatched, err := e.dispatchRequest()
```

Comment:
may I request a brief documentation of the return values of `dispatchRequest()`? Just a one-liner like:
```golang
// dispatchRequest dispatches a subset of requests (selection based on internal heuristic). 
// The boolean return value indicates whether a request was dispatched at all.
```